### PR TITLE
Expose option to set historic feed error callback handler

### PR DESCRIPTION
**Pull request by jleft**
#688

### CODE REVIEW

File: src/assets/js/app.js [hunk 1]
```diff
@@ -309,21 +315,26 @@ export default function() {
...
-                loading(false, 'Error loading data. Please make your selection again, or refresh the page.');
-                var responseText = '';
-                try {
-                    var responseObject = JSON.parse(err.responseText);
-                    var formattedMessage = source.historicNotificationFormatter(responseObject);
-                    if (formattedMessage) {
-                        responseText = '. ' + formattedMessage;
+                if (externalHistoricFeedErrorCallback) {
+                    var error = externalHistoricFeedErrorCallback(err) || true;
+                    loading(false, error);
```
Comment (rjmcneill) [inline src/assets/js/app.js]: If `externalHistoricFeedErrorCallback(err)` isn't set, will this then just display `true` in the notification?

Comment (jleft) [inline src/assets/js/app.js]: No sure exactly what you mean by isn't set? If `externalHistoricFeedErrorCallback` is falsey, then we won't get to this code at all; if it's truthy then we will.

Line 319 is checking the return value of `externalHistoricFeedErrorCallback(err)` - it allows the user to return a string from their callback, which can be set as the error message - if they don't return anything (or return something falsey) then `error` is set to `true`, so the chart will still be obscured by the overlay when there is an error, but no message will be shown.

Comment (rjmcneill) [inline src/assets/js/app.js]: > but no message will be shown

That's the part I wasn't sure about.
I wasn't sure if `true` would get cast to a string or what would happen.

**Comment by rjmcneill** [comment]
Can't +1 on ZenHub atm for some reason, but LGTM

File: src/assets/js/app.js (hunk 1 above)
Comment (jleft) [inline src/assets/js/app.js]: This is the loading function:

```js
function loading(isLoading, error) {
    var spinner = '<div class="spinner"></div>';
    var obscure = arguments.length > 1 || isLoading;

    var errorMessage = '';
    if (error && error.length) {
        errorMessage = '<div class="content alert alert-info">' + error + '</div>';
    }
    containers.app.select('#loading-status-message')
        .classed('hidden', !obscure)
        .html(error ? errorMessage : spinner);
}
```

If `error` is `true`, then this `if (error && error.length)` will evaluate to false, so the message will remain as `''`;

**Comment by jleft** [comment]
Thanks for the reviews guys :)

**Pull request merged**